test/signalrclient-e2e-tests/hub_connection_tests.cpp

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#pragma once
+
+#include "stdafx.h"
+#include <string>
+#include "cpprest\details\basic_types.h"
+#include "cpprest\json.h" 
+#include "connection.h"
+#include "hub_connection.h"
+#include "url.h"
+
+using namespace signalr;
+using namespace utility;
+using namespace std;
+using namespace pplx;
+
+TEST(hub_connection_tests, send_message)
+{
+    auto hubConn = make_shared<hub_connection>(url + U("SignalR"));
+    auto message = make_shared<string_t>();
+    auto received_event = make_shared<pplx::event>();
+    auto test_event = make_shared<pplx::event>();
+
+    auto hubProxy = hubConn->create_hub_proxy(U("hubConnection"));
+
+    hubProxy.on(U("displayMessage"), [message, received_event](const web::json::value& arguments)
+    {
+        *message = arguments.serialize();
+        received_event->set();
+    });
+
+    hubConn->start()
+        .then([hubConn, &hubProxy, test_event](task<void> start_task)
+        {
+            try
+            {
+                start_task.get();
+                ASSERT_TRUE(hubConn->get_connection_state() == connection_state::connected);
```

Comment:
I would check the connection state only in one test... Maybe I would even create a single test that checks connection state and would not pollute other tests.